src/pod-create.ts

```diff
@@ -1,13 +1,190 @@
+  }
+
+  let moduleEnabled = false;
+  modules.forEach(module => {
+    try {
+      // This throws if the address does not match a Controller deployment.
+      getControllerByAddress(module, config.network);
+      // If the above didn't throw, then we've found an Orca module.
+      moduleEnabled = true;
+    } catch {
+      // do nothing
+    }
+  });
+  return moduleEnabled;
+}
+
+/**
+ * Creates a SafeTx on a safe to enable the latest Controller as a module.
+ *
+ * @param safe - Safe address
+ * @param signer
+ * @throws If a Controller module is already enabled. If you are attempting to upgrade versions, use `Pod.migratePodToLatest`.
+ */
+export async function enableController(safe: string, signer: ethers.Signer) {
+  const sender = await signer.getAddress();
+  // Check to see if signer is safe member
+  const { owners, modules } = await getSafeInfo(safe);
+  if (!owners.includes(sender)) throw new Error('Sender was not safe owner');
+
+  if (await isControllerEnabled(modules))
+    throw new Error(
+      "Pod module was already enabled. If you're trying to upgrade versions, use `Pod.migratePodToLatest` instead",
+    );
+
+  const { address: latestModule } = getDeployment('ControllerLatest', config.network);
+
+  let safeTx;
+  try {
+    await createSafeTransaction({
+      sender,
+      safe,
+      to: safe,
+      data: encodeFunctionData('GnosisSafe', 'enableModule', [latestModule]),
+    });
+  } catch (err) {
+    if (err.response?.data.message === 'Gas estimation failed') {
+      throw new Error('Gas estimation failed (this is often a revert error)');
+    }
+    throw err;
+  }
+  await approveSafeTransaction(safeTx, signer);
+}
+
+/**
+ * Adds a Gnosis Safe to the pod ecosystem.
+ * If a signer is not provided, it instead returns the unsigned transaction.
+ * @param args.admin - Optional address of admin
+ * @param args.name - ENS label for safe, i.e., 'orca-core'. Do not add the pod.eth/pod.xyz suffix
+ * @param args.safe - Safe address
+ * @param signer - Signer of a safe owner.
+ * @throws - If Controller module was not enabled
+ * @throws - If signer is not a safe owner
+ * @returns
+ */
+export async function podifySafe(
+  args: {
+    admin?: string;
+    name: string;
+    safe: string;
+  },
+  signer?: ethers.Signer,
+): Promise<ethers.providers.TransactionResponse | { to: string; data: string }> {
+  const { owners, modules } = await getSafeInfo(args.safe);
+  if (signer && !owners.includes(await signer.getAddress()))
+    throw new Error('Sender was not safe owner');
+
+  if (!(await isControllerEnabled(modules))) throw new Error('Pod module was not enabled');
+  // Checksum all addresses
+  const admin = args.admin ? ethers.utils.getAddress(args.admin) : ethers.constants.AddressZero;
+  try {
+    const MemberToken = getContract('MemberToken', config.provider);
+    const expectedPodId = (await MemberToken.getNextAvailablePodId()).toString();
+    const Controller = getContract('ControllerLatest', signer);
```

Comment:
we can't rely on them having enabled latest module, we will probably need to figure out which module they have enabled